File: app/src/main/java/com/example/vlad/financemanager/data/models/Category.java
package com.example.vlad.financemanager.data.models;

import android.text.TextUtils;

import java.io.Serializable;

public class Category implements Serializable {
    private int id;
    private String name;
    private int icon;
    private boolean isCustom;
    private boolean isInputCategory;

    public Category(int id, String name, int icon, boolean isCustom, boolean isInputCategory) {
        this.id = id;
        this.name = name;
        this.icon = icon;
        this.isCustom = isCustom;
        this.isInputCategory = isInputCategory;
    }

    public Category() {
        this(0, "Unknown", 0, true, false);
    }

    public boolean getIsCustom() {
        return isCustom;
    }

    public boolean getIsInputCategory() {
        return isInputCategory;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        if (!TextUtils.isEmpty(name)) {
            this.name = name;
        }
    }

    public int getIcon() {
        return icon;
    }

    public void setIcon(int icon) {
        this.icon = icon;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: app/src/main/java/com/example/vlad/financemanager/data/models/Category.kt
package com.example.vlad.financemanager.data.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Category(var id: Int,
                    var name: String,
                    var icon: Int,
                    var isCustom: Boolean,
                    var isInputCategory: Boolean) : Parcelable {

    constructor() : this(-1, "", -1, false, false)
}
